### DISCUSSION TITLE
Any other DL/HPC related algorithm that CUTLASS could adapted in?

### DISCUSSION THREAD

**Discussion by Peter9606** [NVIDIA/cutlass#181] [General]
Just wondering if there is any other algorithm in deep learning or HPC domains that CUTLASS could fit in.

**Comment by hwu36** [comment]
There are a lot.  Here are those that are on top of my head.

Block sparse: https://github.com/openai/blocksparse

Mos of things here: http://clmathlibraries.github.io/clBLAS/group__BLAS3.html

Semi ring gemm: https://github.com/hpcgarage/cuASR

Others can add more.

**Reply by Peter9606** [reply]
Thanks for sharing and looking forward updates on upstream

**Reply by Peter9606** [reply]
Is it possible for CUTLASS also enable GitHub's "Projects" function just like what CUB does.

**Reply by hwu36** [reply]
Sure.  Please take a look at this doc: https://docs.github.com/en/github/managing-your-work-on-github/creating-a-project-board#creating-a-repository-project-board and let me know what kind of project board do you want us have.

**Reply by Peter9606** [reply]
Personally I'd like to have "Plan your project", as I'm curious about the road map.

**Reply by hwu36** [reply]
I created a blank project.  Feel free to play with it you want.

As to the roadmap, CUTLASS is tightly connected to NVIDIA hardware development.  I will see how much that I can say.

**Reply by Peter9606** [reply]
I'd like to add some open issues as "To Do" in Kanban, but I cannot find "Add Cards" in the project.

**Reply by hwu36** [reply]
I can see add cards in the upper right corner.

**Reply by Peter9606** [reply]
Following is my view. Do we have the same options?

![image](https://user-images.githubusercontent.com/2186779/112401726-20a76300-8d46-11eb-9144-3b20117494f5.png)

**Comment by hwu36** [comment]
if you close menu, you can see "add cards" left to the Menu button.

**Reply by Peter9606** [reply]
I'm afraid I still cannot find it.

![project2](https://user-images.githubusercontent.com/2186779/112403228-002cd800-8d49-11eb-9cdb-2ac6a1e4fa2d.png)

**Comment by hwu36** [comment]
I don't know much about how to use project feature.  I guess only the maintainer can add cards.  Actually a card was created automatically when I created this project, but I don't see it in your screenshot.

**Reply by Peter9606** [reply]
That's alright. 
Actually I'm thinking about adding a couple of "TODO" items to track open issues, some of them are listed here:

- [Profiling parallel split-K gemm](https://github.com/NVIDIA/cutlass/issues/215)
- [Resurrect GEMV](https://github.com/NVIDIA/cutlass/issues/120)
- and more...

**Comment by hwu36** [comment]
Thank you very much.  There are couple more that contributor outside of Nvidia can help us.

- fix all the bugs when building for maxwell and pascal.
- fully support clang.
- clean up the code.
- add "-Werror -Wall"

We are encouraging the community help us ;)

**Reply by Peter9606** [reply]
Hope these tasks can be manged well by utilizing Project tool
